src/sql/workbench/services/query/common/gridDataProvider.ts

```diff
@@ -132,6 +140,67 @@ export async function getResultsString(provider: IGridDataProvider, selection: S
 	return copyString;
 }
 
+export function getTableHeaderString(provider: IGridDataProvider, selection: Slick.Range[], delimiter: string, columns?: Slick.Column<any>[]): string {
+	let headers: Map<number, string> = new Map(); // Maps a column index -> header
+
+	if (selection.length > 0) {
+		selection.forEach((range) => {
+			let startCol = range.fromCell;
+			let columnHeaders = provider.getColumnHeaders(range);
+			if (columnHeaders !== undefined) {
+				let idx = 0;
+				for (let header of columnHeaders) {
+					headers.set(startCol + idx, header);
+					idx++;
+				}
+			}
+		});
+	}
+	else if (columns) { // If a copy selection isn't specified then copy all column headers
+		columns.forEach((column, index) => {
+			// Skipping index 0 since that's the column for the row number
+			if (index === 0) {
+				return;
+			}
+
+			let columnHeader = column.name;
+			if (columnHeader?.includes('&quot;')) {
+				columnHeader = columnHeader.replace(/&quot;/g, '\"');
+			}
+
+			headers.set(index, (columnHeader ? columnHeader : ''));
+		});
+	}
+
+	const sortResults = (e1: [number, any], e2: [number, any]) => {
```

Comment:
It ensures that column headers appear left to right. So if a user clicks on a column somewhere in the middle, then clicks on a column on the left side, and then the right side. Sorting ensures that column headers appear as left, middle, right, instead of middle, left, right, which would have been the order they were clicked on in this example.

Reply:
Please add a comment about that then. Also : 

1. This seems to be reused from the copy results - would suggest a general helper method both can use
2. Also adding a comment to other code as well would be great